src/java.desktop/share/classes/sun/java2d/loops/GraphicsPrimitiveMgr.java

```diff
@@ -274,6 +241,46 @@ private static void writeLog(String str) {
         }
     }
 
+    /**
+     * A holder for general primitives to avoid circular dependencies
+     * between GraphicsPrimitiveMgr and Blit/etc classes.
+     */
+    final static class GeneralPrimitives {
+
+        private static GraphicsPrimitive[] primitives;
+
+        /**
+         * Registers the general loop which will be used to produce specific
+         * primitives by the {@link GraphicsPrimitive#makePrimitive} function.
+         *
+         * @param gen the graphics primitive to be registered as the general loop
+         */
+        static synchronized void register(GraphicsPrimitive gen) {
```

Comment:
The change looks fine, but it seems we have a good opportunity here to change all that synchronized methods to read/write locks.